src/sentry/snuba/metrics/query_builder.py

```diff
@@ -297,7 +298,7 @@ def _parse_offset(self, paginator_kwargs):
         return Offset(offset) if offset else None
 
     def _validate_series_limit(self, query_params):
-        if self.limit:
+        if self.limit and query_params.get("per_page") is not None:
```

Comment:
Added this check to make sure not to run this check when it is the default value for limit as we will override it later

Reply:
I wish we could validate this without checking the `per_page` parameter, but the only way I can think of is silently lowering the limit to `max_points // intervals` instead of raising an error, and I'm not sure if that's a better solution.

Reply:
I thought about that actually but I think it is better to be explicit about it i.e. raising the error here